.github/actions/publish/action.yml

```diff
@@ -28,9 +28,9 @@ runs:
   using: 'composite'
   steps:
     - name: Download custom source artifact
-      uses: actions/download-artifact@v3
+      uses: actions/download-artifact@f44cd7b40bfd40b6aa1cc1b9b5b7bf03d3c67110 # v4.1.0
       with:
-        name: custom-${{ github.event.pull_request.head.sha || github.event.head_commit.id }}-${{ runner.os }}-${{ runner.arch }}
+        name: custom-${{ github.event.pull_request.head.sha || github.event.head_commit.id }}-${{ strategy.job-index }}-${{ runner.os }}-${{ runner.arch }}
```

Comment:
you should omit `${{ strategy.job-index }}` from here because it won't necessarily align with the uploaded index